spec/support/thread_order_support.rb

```diff
@@ -0,0 +1,93 @@
+class ThreadOrderSupport
+  def initialize
+    @bodies  = {}
+    @threads = []
+    @queue   = [] # we may not have thread stdlib required, so may not have Queue class
```

Comment:
The spec that enforces the whitelist of allowed stdlibs loads only the `lib` files of the project, not the `spec` files, so libraries loaded in spec files to facilitiate testing are not considered.